src/utils/ShuffleInfo.java

```diff
@@ -0,0 +1,120 @@
+package utils;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import main.ArrayVisualizer;
+
+final public class ShuffleInfo {
+    final boolean isDistribution;
+    final Distributions distribution;
+    final Shuffles shuffle;
+    final boolean warpDistribution;
+
+    public ShuffleInfo(Distributions distribution, boolean warpDistribution) {
+        this.isDistribution = true;
+        this.distribution = distribution;
```

Comment:
the `CUSTOM` distribution is not properly initialized

Reply:
This would be extremely difficult, as the `CUSTOM` distribution is implemented as a singleton. Just don't use it as a non-base distribution for now.

Reply:
in that case, you should probably remove it from the lists

Reply:
That's not easily possible

Reply:
can't you just filter the word "Custom" from the list?

Reply:
Not easily, because of the stupid way `ListModel`s work.

Reply:
I tried doing it myself and it seems to work fine?

Reply:
I wouldn't consider that "easily" lol (and did you do it for `jList2` as well?)

Reply:
I'd consider it pretty "easily" (given that these are static arrays), this was just a demo

Reply:
This also doesn't work, because it assumes that the indices of elements in the JList will be 1:1 with the indices in the enum. (so if anybody adds any shuffles after `CUSTOM`, it will break completely)

Reply:
why not access the distributions using the string keys?

Reply:
It's more complicated, so it can be done, but I'll only use that where necessary.

Reply:
couldn't you just use `getSelectedValue()`, do a null check, and access the distributions directly?

Reply:
Access the distributions directly?